integrationtests/testground/main.go

```diff
@@ -0,0 +1,453 @@
+// This is the test program to be run by testground. Here's what it does:
```

Comment:
The code logic is still a bit obscure, but that is the result of several iterations trying to make it more understandable, so I add this overview to help.